### PR TITLE
Add validateTaskProperties task to buildSrc

### PR DESCRIPTION
**Pull request by wolfs**

### CODE REVIEW

File: buildSrc/build.gradle.kts [hunk 1]
```diff
@@ -71,6 +71,21 @@ subprojects {
     dependencies {
         compile(gradleApi())
     }
+
+    afterEvaluate {
+        if (tasks.withType<ValidateTaskProperties>().isEmpty()) {
```
Comment (lptr) [inline buildSrc/build.gradle.kts]: Can we not have the `if`? We know that we don't have such a task, so we should always create it.

Comment (wolfs) [reply buildSrc/build.gradle.kts]: Some subprojects apply the `java-gradle-plugin`, so they already have the `validateTaskProperties` task.

Comment (lptr) [reply buildSrc/build.gradle.kts]: Can we apply it everywhere and make plugins the business of a dedicated `gradle-plugin-plugin` perhaps? It seems that we are capturing the wrong thing here by expecting tasks to be only be part of plugins.

Comment (wolfs) [reply buildSrc/build.gradle.kts]: I think we should split the `java-gradle-plugin` into two plugins - one doing adding the validation task (and other task quality things maybe) and another one dealing with the work for packaging the plugin correctly (your `gradle-plugin-plugin`).

That requires some work on Gradle itself. Until then, I don't have a better solution that to check if the task already exists.

Is this what you meant?

Comment (lptr) [reply buildSrc/build.gradle.kts]: I think that's fair. Would you mind creating an issue to split the plugin?

**Pull request merged**